Not building on OpenBSD

Fails with:

```
error[E0425]: cannot find function `errno_location` in this scope
  --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/daemonize-0.2.3/src/
   |
52 |     *errno_location()
   |      ^^^^^^^^^^^^^^ not found in this scope
```

--- a/src/ffi.rs
+++ b/src/ffi.rs
@@ -42,7 +42,7 @@ unsafe fn errno_location() -> *const libc::c_int {
     extern { fn __errno_location() -> *const libc::c_int; }
     __errno_location()
 }
-#[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd"))]
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd", target_os = "openbsd"))]
 unsafe fn errno_location() -> *const libc::c_int {
     extern { fn __error() -> *const libc::c_int; }
     __error()

--- a/src/ffi.rs
+++ b/src/ffi.rs
@@ -42,7 +42,15 @@ unsafe fn errno_location() -> *const libc::c_int {
     extern { fn __errno_location() -> *const libc::c_int; }
     __errno_location()
 }
-#[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd", target_os = "openbsd"))]
+
+
+#[cfg(target_os = "openbsd")]
+unsafe fn errno_location() -> *const libc::c_int {
+    extern { fn __errno() -> *const libc::c_int; }
+    __errno()
+}
+
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "freebsd"))]
 unsafe fn errno_location() -> *const libc::c_int {
     extern { fn __error() -> *const libc::c_int; }
     __error()

--- a/src/ffi.rs
+++ b/src/ffi.rs
@@ -95,6 +95,11 @@ mod tests {
     unsafe fn nobody_uid_gid() -> libc::uid_t {
         (u32::max_value() - 1) as libc::uid_t
     }
+    
+    #[cfg(target_os = "openbsd")]
+    unsafe fn nobody_uid_gid() -> libc::uid_t {
+        (i16::max_value()) as libc::uid_t
+    }
 
     #[test]
     fn test_get_gid_by_name() {

